src/GoogleAdWordsEventForwarder.js

```diff
@@ -124,23 +120,46 @@
             return 'Can\'t send to forwarder ' + name + ', not initialized. Event added to queue.';
         }
 
-        function setEnhancedConversionData(enhancedConversionData) {
-            if (enhancedConversionData.email) {
-                window.enhanced_conversion_data.email = enhancedConversionData.email;
-            }
-            if (enhancedConversionData.phone_number) {
-                window.enhanced_conversion_data.phone_number = enhancedConversionData.phone_number;
-            }
-            if (enhancedConversionData.first_name) {
-                window.enhanced_conversion_data.first_name = enhancedConversionData.first_name;
-            }
-            if (enhancedConversionData.last_name) {
-                window.enhanced_conversion_data.last_name = enhancedConversionData.last_name;
-            }
-            if (enhancedConversionData.home_address) {
-                window.enhanced_conversion_data.home_address =
-                    enhancedConversionData.home_address;
+        function sanitizeEnhancedConversionData(conversionData) {
+            const sanitizedConversionData = {};
+
+            const allowedKeys = [
+                'email',
+                'phone_number',
+                'first_name',
+                'last_name',
+                'home_address',
+            ];
+
+            allowedKeys.forEach(function (key) {
+                if(conversionData[key]) {
+                    sanitizedConversionData[key] = conversionData[key];
+                }
+            });
+
+            return sanitizedConversionData;
+        }
+
+        function hasEnhancedConversionData(customFlags) {
+            return customFlags && Object.keys(customFlags).length && customFlags[ENHANCED_CONVERSION_DATA];
+        }
+
+        function parseEnhancedConversionData(conversionData) {
+            // Checks if conversion data in custom flags is a stringified object
+            // Conversion data should only be a stringified object or JSON
+            if (typeof conversionData === 'string') {
+                try {
+                    return JSON.parse(conversionData);
+                } catch (error) {
+                    console.warn('Unrecognized Enhanced Conversion Data Format', conversionData, error);
+                    return {};
+                }
+            } else {
+                // Not a string, no need to sanitize, but we should not
```

Comment:
This comment makes it sound like sanitization isn't happening in this code block, but is in the previous if block on line 150-156, but it isn't happening there either. 

Sanitizing happens regardless if it's a string or an object, on line 59, which happens after `parseEnhancedConversionData` is called in line 58, so this comment is misleading.